lib/db/schema/patch-041-042.sql

```diff
@@ -0,0 +1,24 @@
+-- This migration creates the `deviceFromTokenVerificationId_1`
+-- procedure that tries to find a device associated with a session
+-- that can be verified with a specific tokenVerificationId.
+
+CREATE PROCEDURE `deviceFromTokenVerificationId_1` (
+    IN inUid BINARY(16),
+    IN inTokenVerificationId BINARY(16)
+)
+BEGIN
+    SELECT
+        d.id AS id,
+        d.name AS name,
+        d.type AS type,
+        d.createdAt AS createdAt,
+        d.callbackURL AS callbackURL,
+        d.callbackPublicKey AS callbackPublicKey,
+        d.callbackAuthKey AS callbackAuthKey
+    FROM devices AS d
+    LEFT JOIN unverifiedTokens AS u
```

Comment:
Using a`LEFT JOIN` here is awkward, because you *must* match a row in both tables for the query to return any rows.  Since you're finding the row in `u` and then matching to a row in `d`, I suggest putting them in the opposite order for clarity and using an `INNER JOIN`:

`FROM unverifiedTokens u INNER JOIN devices d ON ON u.tokenId = d.sessionTokenId AND u.uid = d.uid`